Lib/linecache.py

```diff
@@ -135,10 +139,17 @@ def updatecache(filename, module_globals=None):
             try:
                 stat = os.stat(fullname)
                 break
-            except OSError:
+            except (OSError, ValueError):
                 pass
         else:
             return []
+    except ValueError:
+        # ValueError may happen on Windows platforms for long paths or
```

Comment:
It also happen when the file name is not encodable. If it contains surrogate codes or the locale is not UTF-8.

I suggest to not enumerate all possible variants of the failure. Just say why the code does what it does if this is not obvious. Repetitive long comments only distract.

There are no long comments for error handlers in `genericpath.py` (well, I the author of this code, so this is not surprising).